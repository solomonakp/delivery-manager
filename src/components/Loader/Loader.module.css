.loaderContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: inherit;
  z-index: 100;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  position: absolute;
}

.fixed {
  position: fixed !important;
}

.ldsRoller {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}
.ldsRoller div {
  -webkit-animation: lds-roller 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  animation: lds-roller 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  transform-origin: 40px 40px;
}
.ldsRoller div:after {
  content: ' ';
  display: block;
  position: absolute;
  width: 7px;
  height: 7px;
  border-radius: 50%;

  margin: -4px 0 0 -4px;
}
.ldsRoller div:nth-child(1) {
  -webkit-animation-delay: -0.036s;
  animation-delay: -0.036s;
}
.ldsRoller div:nth-child(1):after {
  top: 63px;
  left: 63px;
}
.ldsRoller div:nth-child(2) {
  -webkit-animation-delay: -0.072s;
  animation-delay: -0.072s;
}
.ldsRoller div:nth-child(2):after {
  top: 68px;
  left: 56px;
}
.ldsRoller div:nth-child(3) {
  -webkit-animation-delay: -0.108s;
  animation-delay: -0.108s;
}
.ldsRoller div:nth-child(3):after {
  top: 71px;
  left: 48px;
}
.ldsRoller div:nth-child(4) {
  -webkit-animation-delay: -0.144s;
  animation-delay: -0.144s;
}
.ldsRoller div:nth-child(4):after {
  top: 72px;
  left: 40px;
}
.ldsRoller div:nth-child(5) {
  -webkit-animation-delay: -0.18s;
  animation-delay: -0.18s;
}
.ldsRoller div:nth-child(5):after {
  top: 71px;
  left: 32px;
}
.ldsRoller div:nth-child(6) {
  -webkit-animation-delay: -0.216s;
  animation-delay: -0.216s;
}
.ldsRoller div:nth-child(6):after {
  top: 68px;
  left: 24px;
}
.ldsRoller div:nth-child(7) {
  -webkit-animation-delay: -0.252s;
  animation-delay: -0.252s;
}
.ldsRoller div:nth-child(7):after {
  top: 63px;
  left: 17px;
}
.ldsRoller div:nth-child(8) {
  -webkit-animation-delay: -0.288s;
  animation-delay: -0.288s;
}
.ldsRoller div:nth-child(8):after {
  top: 56px;
  left: 12px;
}

@-webkit-keyframes lds-roller {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
@keyframes lds-roller {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.ldsEllipsis {
  display: block;
  position: relative;
  width: 80px;
  height: 80px;
}

.ldsEllipsis span {
  display: inline-block;
  position: absolute;
  top: 33px;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  -webkit-animation-timing-function: cubic-bezier(0, 1, 1, 0);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.ldsEllipsis span:nth-child(1) {
  left: 8px;
  -webkit-animation: lds-ellipsis1 0.6s infinite;
  animation: lds-ellipsis1 0.6s infinite;
}
.ldsEllipsis span:nth-child(2) {
  left: 8px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}
.ldsEllipsis span:nth-child(3) {
  left: 32px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}
.ldsEllipsis span:nth-child(4) {
  left: 56px;
  -webkit-animation: lds-ellipsis3 0.6s infinite;
  animation: lds-ellipsis3 0.6s infinite;
}

@-webkit-keyframes lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@-webkit-keyframes lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@-webkit-keyframes lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(24px, 0);
  }
}
@keyframes lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(24px, 0);
  }
}
