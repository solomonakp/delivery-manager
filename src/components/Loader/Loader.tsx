import React, { FC } from 'react';
import styles from './Loader.module.css';

export interface LoaderProps {
  type?: 1 | 2;
  fixed?: boolean;
  className?: string;
}

const Loader: FC<LoaderProps> = ({ type = 1, fixed = false, className }) => {
  return (
    <div
      className={`${styles.loaderContainer} ${className} ${
        fixed ? styles.fixed : ''
      }`}
      data-testid='loader'
    >
      {type === 1 ? (
        <div className={`${styles.ldsRoller} bg`}>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
          <div className='after-color-1'></div>
        </div>
      ) : (
        <span className={`${styles.ldsEllipsis}`}>
          <span className='after-color-2'></span>
          <span className='after-color-2'></span>
          <span className='after-color-2'></span>
          <span className='after-color-2'></span>
        </span>
      )}
    </div>
  );
};

export default Loader;
