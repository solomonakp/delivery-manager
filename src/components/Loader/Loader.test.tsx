import React from 'react';
import { render, screen } from '@testing-library/react';
import Loader, { LoaderProps } from './Loader';

const setUpLoader = (props?: LoaderProps) => render(<Loader {...props} />);

describe('<Loader />', () => {
  test('it should mount', () => {
    setUpLoader();
    expect(screen.getByTestId('loader')).toBeInTheDocument();
  });
  test('should render the Loader user would see', () => {
    const { asFragment } = setUpLoader();

    expect(asFragment()).toMatchSnapshot();
  });
});
