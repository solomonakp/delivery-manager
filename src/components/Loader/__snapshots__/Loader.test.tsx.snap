// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the Loader user would see 1`] = `
<DocumentFragment>
  <div
    class="loaderContainer undefined "
    data-testid="loader"
  >
    <div
      class="ldsRoller bg"
    >
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
      <div
        class="after-color-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
